### ISSUE TITLE
Insecure deserialization active scanner

### ISSUE THREAD

**Issue by omerlh** [zaproxy/zaproxy#4112]
As already discussed in #3883, this is now part of the new Top 10 - A8. Could be cool to add such an active scan rule - based on this [tool](https://github.com/pwntester/ysoserial.net) could be a good start for .NET...

**Comment by kingthorin** [comment]
Other projects or info:

- https://github.com/federicodotta/Java-Deserialization-Scanner (Includes vuln apps.)
- Java version of yoserial: https://github.com/frohoff/ysoserial

**Comment by kingthorin** [comment]
https://github.com/pwntester/ysoserial.net

**Comment by sixsec** [comment]
Another Project that could maybe be ported:
https://github.com/nccgroup/freddy

**Comment by NF997** [comment]
Has anyone already started to work on this?

**Comment by thc202** [comment]
I don't think so.

**Comment by ssyms** [comment]
I would potentially be interested in this if it's still open? I couldn't find an insecure deserialization rule in the Active Scan docs, but just want to be sure

**Comment by kingthorin** [comment]
This is definitely still available.

**Comment by ssyms** [comment]
Awesome, will start looking into it

**Comment by pranavsaxena17** [comment]
@ssyms Are you still on this? I would like to look into this.

**Comment by ssyms** [comment]
@pranavsaxena17 Yes, I've been a bit busy since the New Year but I have done some work on it and would like to finish.

**Comment by jangelesg** [comment]
Id like to start working on this one; any suggestion I'm new to the team 

Greetings

**Comment by ssyms** [comment]
Hi Jon, I am done implementing functionality for Java deserialization, just working on adding tests. Maybe you could look into integrating the .NET tool? (https://github.com/pwntester/ysoserial.net)

**Comment by jangelesg** [comment]
> Hi Jon, I am done implementing functionality for Java deserialization, just working on adding tests. Maybe you could look into integrating the .NET tool? (https://github.com/pwntester/ysoserial.net)

I will

**Comment by ricekot** [comment]
@ssyms @jangelesg how are things going? do you need any help?

**Comment by ssyms** [comment]
@ricekot I'm open to this ticket being re-assigned. A lot of personal stuff has come up in the past year and I'm still busy dealing with it.

**Comment by Ahxius** [comment]
Is this still available? I'd love to work on this – I haven't contributed before, but I should have what it takes to get a working solution.

**Comment by kingthorin** [comment]
Go for it.
Dev info here: https://www.zaproxy.org/docs/developer/ 😃
